library/lua/dfhack.lua

```diff
@@ -74,10 +80,16 @@ end
 safecall = dfhack.safecall
 curry = dfhack.curry
 
+---@generic T
+---@param f fun(...):T
+---@param ... any
+---@return T
 function dfhack.pcall(f, ...)
```

Comment:
this one is tricky. On success, this function returns `true, ...` where `...` are the return values from `f` (different from the `@param ...`, which are the arguments to `f`).
On failure, it returns `false, err` where err is the error object, which I think is always string error message, at least whenever I've seen it used.

Reply:
Probably a case where we run against the edge of what LuaLS can actually represent. I think in this case we just do:

```
---@generic T
---@param f fun(...): T
---@param ... any -- LuaLS does not like typing variable arguments and especially not inside of a inline function :-(
---@return boolean
---@return T|string ... -- LuaLS will not correctly type variable return types as T but this is not a major drawback.
``

Reply:
lgtm